### DISCUSSION TITLE
Organizing shared photos without duplicates

### DISCUSSION THREAD

**Discussion by UnderTheCarpet** [ente-io/ente#790] [Enhancements]
I cannot assign an individual album to a photo shared with me. In fact, no organisational feature seems to work. This is not only confusing, but IMHO a sever limitation in the user experience. This makes family sharing quite hairy.

I'd like to suggest that photos shared with me should behave just like my own photos.

Perhaps a duplicate of ente-io/photos-app#1401?

**Comment by vishnukvmd** [comment]
Hey, yes, the underlying issue for ente-io/photos-app#1401 is the same. But I'll keep this issue open as well, since it does a better job of documenting the behavior expected out of shared data.

This is very much on our radar, just that we're currently focused on delivering Search. Once that's done, we'll get back to ironing out the Sharing flows.

**Reply by UnderTheCarpet** [reply]
It seems this is implemented in the current release? Nice! And thank you.

However, it seems to create an independent copy of the file (which is a sensible first approach). This is a bit of a pity and goes against a collaborative spirit. That is, what about adding metadata? Now both have do the same work. Not to mention that now an auxiliary copy is created counting against the storage.

Furthermore, it is plain weird: Let's say my friend shares an album with me. If I now heart a shared photo from that album and readd it to the album now the same photo is twice in the album (mine and my friend's). I think this is bad UX when a photo looses its unique identity.

I can imagine why you chose this way (because it is easier). However, from a technical point of view it is weird that putting a photo into an album now creates a copy for shared items when all is needed is a reference.

I'm afraid to use this feature because it will fragment all my shared photos and I fear this cannot be cleanup (and merged) later. I also fear my peers will use this and create a myriad of independent copies of the same photo. This quickly becomes unmanageable!

@vishnukvmd Are there plans to make sharing, organising and sharing photos truly collaborative in the sense that the same copy is shared?

**Reply by shirazkmz** [reply]
I'd like to request that this feature have a flag to optionally disable on a per-album basis. In my case, I share albums with Viewers (not collaborators) and would like them not to be able to "heart" a photo and have it downloaded into their own accounts. My goal is to share family photos that I can limit distribution of and this effectively pulls photos out of my circle of control (where before I could simply remove a viewer, now they'll have a copy of a photo even if removed).

**Reply by UnderTheCarpet** [reply]
@shrkmz I'd argue that once you have shared a photo it must be considered out of your control because it is always possible to make a copy regardless.

**Reply by shirazkmz** [reply]
I don't necessarily share that perspective here, and unfortunately "always possible to make a copy regardless" is a straw man argument here IMO. A couple of examples that illustrate my thought process:

- Signal messaging has the option of deleting media for everyone in a message thread. An example of implemented functionality with the intent of allowing the original sender to remove data if they desired. The argument can be made that someone on the other end can take a screenshot but that is exactly the straw man argument.
- Google Photos has the option of removing a collaborator from an album and thereby removing their access to the photos: https://support.google.com/photos/answer/9789702?hl=en#zippy=%2Chow-we-protect-your-shared-albums%2Cyoure-in-control

One more is Proton Drive's sharing, while they don't have the same feature set implemented as Ente does, it's clear the goal is to ensure the owner has as much control as possible on what is shared and when https://proton.me/blog/share-photos-online-privately

I think the album owner should have control over whether participants can download media from an album they own. Having the platform default to allowing external parties to download with no ability to disable is a privacy anti-pattern. Tbh, I'd consider not using the album sharing feature at all if I know sharing once means I may never be able to remove someone I invited once from access to that media.

**Reply by ua741** [reply]
> Google Photos has the option of removing a collaborator from an album and thereby removing their access to the photos:

You can do the same within Ente Photos as well. 
For what it's worth, in the case of Google Photos or any other photo storage app that provides similar functionality, once a participant has copied the photos to their personal storage space, the copies remain intact even if the original uploader deletes the photos. 
If I remember correctly, in Signal, if a person forwards a media file, the forwarded media copy remains intact.

**Reply by shirazkmz** [reply]
I do understand that a copy can be made in any other app manually. The previous configuration in Ente where a Viewer would have to make a conscious choice to forward or download media and therefore make a copy outside the album owners control was perfect though.

The Heart/Favorite now does this without clear information for the Viewer, and there's no way to disable this by the album owner.

I think referencing the fact that media can be forwarded and a screenshot be taken and so on is just a straw-man argument. Respectfully, Ente's goal is to be privacy-focused as an alternative to Google Photos, so I'd expect it to be in line with the mission to provide the content owner with as much control as possible, and the Viewer with as many guardrails as possible, to prevent data leaking.

**Reply by aleqx** [reply]
Has there been any progress on this?

I also won't be using the sharing features without guardrails. I fully agree that "it can be obtained anyway in some form" is not just a strawman agreement but goes counter to the very message of Ente.

The whole point is to make it harder, not easier, for the majority to obtain a copy.

**Reply by willtonkin** [reply]
I want to collaborate on an album with my partner. They are not allowed to heart/favorite or organize the photos added to that album which causes frustration on their part. I'd expect them to be able to enjoy the photos I share with them in the same manor as I can. As I with the photos they add to that same album.

I understand the desire for "guardrails", and yes privacy is a core tenant of the platform. I would also like to point out the rest of Ente Photo's statement "While security and privacy form the bedrock of Ente Photos, it is not at the cost of usability." -- So I'd want to also ensure that with that we also uphold the "usability" part that we and our families also enjoy ❤️ (if I can't convince my relatives to use secure stuff, I'm back to whatever I can get them to use 😅 which won't be as secure and private).

**Reply by shirazkmz** [reply]
@willtonkin , absolutely, I wouldn't want to use Ente if it wasn't usable and your exact use case is mine as well. However this seems like a simple addition of role-based permissions to allow certain people more generous access than others.

Although the statement that you quoted actually gives me more pause than anything else. "Privacy but not at the cost of usability" is pretty subjective.

**Reply by UnderTheCarpet** [reply]
@shrkmz You seem to feel very strongly about this. I understand your viewpoint, a form of defense in depth.

However, I'd like to see collaboration actually work without surprises first (technical breakthrough) and then add the feature you are looking for. Currently, the feature you desire is basically an unintented technical limitation.

So your suggestion to introduce an "ephemeral read-only sharing" is another feature request in my humble opinion.

Another scenario that is sadly limited for now: If a friend would share photos of a trip together with me and shares the photos with me on it and later the friend decides to unshare or even delete the photos, I'd be pretty baffled to have lost those photos. So I'd like to see a copy on unshare feature - which seems to be the exact opposite of what you are asking.

You see, there are many use cases that should not, in my opinion, be hindered by an implementation detail.

**Reply by shirazkmz** [reply]
I wouldn't say I feel "very" strongly about this considering my risk profile is low. However I also don't classify fundamental security and privacy as "features". Ensuring collaboration works before security happens is kind of... Not secure by design.

The second case raised, someone unsharing an album and the recipient losing access to the photos... Isn't that the point? Someone has decided they don't want to share photos with a recipient anymore, they should be able to unshare yes?

If I shared a document and then unshared it, I would expect the recipient to no longer natively have a copy of that document... I wouldn't even expect them to get notified that the document is no longer shared.

**Reply by UnderTheCarpet** [reply]
> However I also don't classify fundamental security and privacy as "features". Ensuring collaboration works before security happens is kind of... Not secure by design.

This feels like a rather extreme stance to me. To exaggerate: What is the safest car in the world? One that does not drive.

> The second case raised, someone unsharing an album and the recipient losing access to the photos... Isn't that the point? Someone has decided they don't want to share photos with a recipient anymore, they should be able to unshare yes?
>
> If I shared a document and then unshared it, I would expect the recipient to no longer natively have a copy of that document... I wouldn't even expect them to get notified that the document is no longer shared.

I was sketching a different real world use case. And I disagree with your assessment because I reject this narrow definition of sharing. I think you made your wish more than clear that sharing for you should be more of a borrow in this context.

**Reply by mlell** [reply]
> The second case raised, someone unsharing an album and the recipient losing access to the photos... Isn't that the point? Someone has decided they don't want to share photos with a recipient anymore, they should be able to unshare yes?

Having just stumbled on this discussion as I am in the same situation (cannot collaborate  on a shared album with my partner/family), I disagree. There are many reasons why photos are unshared by an individual and many do not mean that they want to revoke access by others. For example:

* Alice meets Bob and his child at a playground, takes some photos of them, shares them, but after a while deletes them because she made them for Bob and does not want them herself
* Alice shares a holiday album to Bob and later notices that there are too many redundant photos inside, so she trashes many. However, she is happy that Bob likes some of them and does not want them gone for him, only for herself
* Alice trashes a photo without remembering all the people she has shared it with. 

I think that the "feature" to revoke access is no gain in privacy and security but compromises on usability. It has therefore only downsides. I do not think this is a straw-man argument, as stated earlier. Just because Signal and others pretend to be able to protect data in this way, it does not mean that it is possible. So I think this argument is itself a faulty appeal to authority. Shared photos can be copied by the recipient, and it is even done automatically by ente if the backup feature is enabled.

**Comment by uhkeller** [comment]
I just stumbled across this as well, as I suspect many will at this time of year. The whole family uses Ente and we have lots of shared albums. Now it's time to pick photos for the annual calendar and I expected Ente to be perfect for this. But to my surprise it isn't. I tried adding a photo shared with me to the Calendar album and… Ente created a duplicate! I now have this photo twice in my gallery. Since if I'm not mistaken Search is pretty much done now, maybe it's time to revisit this?

**Comment by nijel** [comment]
I'd also like to be able to grant all permissions when sharing. Including removal of photos. We often take photos using multiple devices (and Ente accounts), which are automatically backed up by Ente. These folders are shared within the family. Being able to consolidate photos from multiple accounts in a single album is what we would like to achieve—remove unwanted ones and move wanted ones to an album.

**Comment by limbozord** [comment]
I'd be very thankful to know if there are any updates on this feature. If we are getting a sharing system where we don't end up creating duplicates by accident while also ensuring we get to keep the photos if the original uploader decides to delete them to clean up their own photos collection.

**Comment by etvt** [comment]
Adding a concrete data point from daily use: shared albums work for viewing, but the copy-on-organize design takes much of the benefit out of collaborating on family photos — the parts that made me sign up are either missing or painful in practice.

Setup: paid plan, storage shared with a family member. I own an album, I uploaded photos to it, the other person uploaded photos into the same album. Ente Photos on Android 1.3.59, Pixel 7. I only tried these features now, so I can't say whether older versions behave differently.

What I observe when I favorite (or add to one of my own albums) a photo that the other person uploaded:

- A separate copy is created. The instance sitting in the shared album is *not* the one that gets favorited/added — it stays untouched and knows nothing about the new album or favorite.
- The copy loses its Ente-side state: album assignments and processing/ML data. It gets reprocessed from scratch shortly after.
- The two are effectively invisible to each other. "Remove duplicates" does not surface them as a duplicate pair, so there is no way to clean this up afterwards.
- None of this is disclosed. There is no dialog, no hint that organizing a received photo means creating your own copy.

The UI actively suggests the opposite:

- After favoriting, a white heart appears on the shared instance in the album. That heart is misleading — opening that instance shows it is not favorited, and filtering the album to favorites only does not list it, because what was actually favorited is the invisible copy.
- Smaller, related: in multi-select, the Favorite action only appears if at least one of *my own* photos is in the selection. Selecting only photos uploaded by the other person makes the action disappear entirely.

Open question I could not answer myself: does the copy count against storage? If it does, then on a shared/family plan every heart silently consumes quota twice, which makes this considerably more serious than a UX wrinkle.

Why this matters in practice: received photos end up effectively view-only — I cannot favorite them meaningfully, and I cannot collect a selection of the family's photos into an album of my own (e.g. to cast a preselection to a TV) without silently fragmenting the library into copies that nothing can merge back.

Even if the underlying design stays as it is for now, two things would help a lot in the meantime:

1. Tell the user. A clear message when an action on a received photo requires creating a copy, ideally with a way to cancel.
2. Don't show state that isn't real. The heart on the shared instance, and the album's favorites filter, should agree with each other.
